### DISCUSSION TITLE
Attempting to port Keyball39 to QMK Firmware 0.25.x (Help Wanted)

### DISCUSSION THREAD

**Discussion by noroadsleft** [Yowkees/keyball#604] [Q&A]
Greetings,

I am noroadsleft, and I am one of the maintainers of QMK Firmware.

Recently, a user came to us via our Discord server requesting QMK Configurator support for the Keyball39. As QMK Configurator only supports keyboards whose support is merged into [the main QMK repository's `master` branch](https://github.com/qmk/qmk_firmware/tree/master/), I offered to investigate merging support.

I've been working with this user over the last few days and am stumped with something. Any firmware I compile for the Keyball39 from the source I've worked out ends up with a reversed keymap when flashed to their board (POIUYTREWQ instead of QWERTYUIOP; trackball works in reverse), but when flashed with the firmware present in [Remap](https://remap-keys.app/catalog/k895xiCpsM5zlYZCPTEN/firmware) it works as intended.

I was hoping to get a sanity check for my code from someone who truly knows the hardware. I've been working on this for several days now, and have yet to figure out why my source compiles in such a manner. Assistance in this matter would be much appreciated.

My source code is here:

- https://github.com/noroadsleft/qmk_firmware/tree/87974d3e769fca4ed71b54a234e4b50cabd1a761/keyboards/yowkees/keyball39
  - branch name: `kb/yowkees_keyball39-0.25.x-full`
  - QMK version: 0.25.6 ([compare](https://github.com/qmk/qmk_firmware/compare/0.25.6...noroadsleft:qmk_firmware:kb/yowkees_keyball39-0.25.x-full))

:spiral_notepad: The path to the keyboard source is `keyboards/yowkees/keyball39/` instead of `keyboards/keyball/keyball39/` but otherwise should be functionally identical to the code from [the Yowkees repository as of Monday (United States Pacific Time)](https://github.com/Yowkees/keyball/tree/ef172dd614839034429452b0f41a343fe004ad8e/qmk_firmware/keyboards/keyball/keyball39). I did originally attempt to compile against QMK Firmware 0.22.14 per the build guide, but ran into a couple of compile issues (fixed in the second and third commits of the branch; the error messages are included in the Git log). If I can get this code operational, my intention is to merge it to QMK mainline.

**Comment by okuRaku** [comment]
related:

https://github.com/Yowkees/keyball/issues/495

https://github.com/Yowkees/keyball/issues/596

(I, too, was needing the features of newer QMK but decided implementing them myself would be quicker than trying to understand what was necessary to migrate)

**Reply by noroadsleft** [reply]
#495 might be it. I forgot that the `SPLIT_HAND_MATRIX_GRID` behaviour was changed.

Also didn't think to search Issues and not just Discussions. :man_facepalming:
